src/core/utils/wrap_stdbool.h

```diff
@@ -44,7 +44,7 @@
 /* c++ has a built in bool */
 #else
 
-#if defined(_MSC_VER)
+#if HAVE__BOOL
```

Comment:
I agree with @nibanks - this is not a trivial thing.
Does this change pass all appvayor tests?

I had many problems like this because of the way VS defines Bool, BOOL and _Bool.

The second problem is this: VS projects are built with *TWO* different versions of Visual Studio.
Technically - this means VS should provide *TWO* different copies of  "openthread-windows-config.h" - one for VS(version-X) and a different one for VS(version-Y).

Why two? Because VS(2015) is C99 compliant... and kernel build is done with an *OLDER* version of Visual Studio.